features/facebookImport/src/components/activitiesMinistory/activitiesMinistory.jsx

```diff
@@ -0,0 +1,167 @@
+import React, { useState } from "react";
+
+import i18n from "../../i18n";
+
+import BarChartHorizontal from "../dataViz/barChartHorizontal.jsx";
+
+import { fillArray } from "../dataViz/utils.jsx";
+
+import "./activitiesMinistory.css";
+import "./datePicker.css";
+import "../tabs/tabs.css";
+
+const monthsAbbreviation = i18n.t("common:months.abbreviation").split(" ");
+
+//TODO let this support actual dates not only years
+const Datepicker = ({ year, yearRange, onYearChange }) => {
+    const rangeIndex = yearRange.indexOf(year);
+    return (
+        <div className="datepicker">
+            {rangeIndex != 0 ? (
+                <button
+                    className="arrow left"
+                    onClick={() => onYearChange(yearRange[rangeIndex - 1])}
+                >
+                    <img src="./images/angle-left.svg" alt="arrow-left" />
+                </button>
+            ) : (
+                <div className="arrow filler"></div>
+            )}
+            <div className="year">{year}</div>
+            {rangeIndex != yearRange.length - 1 ? (
+                <button
+                    className="arrow right"
+                    onClick={() => onYearChange(yearRange[rangeIndex + 1])}
+                >
+                    <img src="./images/angle-right.svg" alt="arrow-right" />
+                </button>
+            ) : (
+                <div className="arrow filler"></div>
+            )}
+        </div>
+    );
+};
+
+const ActivitiesMinistory = ({ totalEvents }) => {
+    const yearRange = fillArray(Object.keys(totalEvents.values));
+
+    const yearlyTotals = Object.fromEntries(
+        yearRange.map((year) => [
+            year.toString().substring(2, 4),
+            totalEvents.values[year]?.total || 0,
+        ])
+    );
+
+    const [selectedYear, setSelectedYear] = useState(
+        yearRange[yearRange.length - 1]
+    );
+
+    const monthlyTotals = Object.fromEntries(
+        new Array(12)
+            .fill(undefined)
+            .map((_, index) => [
+                monthsAbbreviation[index],
```

Comment:
Have you tried here to use a map on `monthsAbbreviation`? You already have the 12 elements, as well as the values...

Reply:
Yeah a leftover from doing it differently before, nice spot 🥇